backend/entityservice/cache/active_runs.py

```diff
@@ -0,0 +1,49 @@
+"""
+Cache information about the run state in redis.
+
+Note across the cache we use a common redis hash for each run,
+this module uses the 'state' key.
+"""
+from enum import Enum
+import structlog
+
+from entityservice.cache.connection import connect_to_redis
+from entityservice.cache.helpers import _get_run_hash_key
+
+logger = structlog.get_logger()
+
+
+class RunState(Enum):
+    MISSING = None
+    ACTIVE = b'active'
+    DELETED = b'deleted'
+
+
+def _set_run_state(run_id, state):
+    logger.debug(f"Setting run state to '{state.name}'", run_id=run_id)
+    r = connect_to_redis()
+    key = _get_run_hash_key(run_id)
+    return r.hset(key, 'state', state.value)
+
+
+def _get_run_state(run_id):
+    r = connect_to_redis(read_only=True)
+    key = _get_run_hash_key(run_id)
+    # hget returns None if missing key/name, and bytes if present
+    maybe_state = r.hget(key, 'state')
+    logger.debug("Loaded run state from cache", run_id=run_id, state=maybe_state)
+    return RunState(maybe_state)
+
+
+def set_run_state_deleted(run_id):
+    return _set_run_state(run_id, state=RunState.DELETED)
+
+
+def set_run_state_active(run_id):
+    return _set_run_state(run_id, state=RunState.ACTIVE)
+
+
+def is_run_active(run_id):
+    return RunState.ACTIVE == _get_run_state(run_id)
```

Comment:
If the run state is missing from the cache, shouldn't we check the information from the database?
Which leads to another question: how long are we keeping this information in redis? Shouldn't we clean the inactive keys? Should we set an expiry time for the keys?

Reply:
I'm happy for redis to be the source of truth - after all the entire job is screwed if we lose our celery queue which is stored in Redis. In any case such a check wouldn't be here in the `cache.py` file.

Sure we should we delete keys from redis at some point, not entirely sure it make sense now though. Given we are talking about a handful of bytes per run and by default we provision redis-ha with 1.5GiB of memory... I don't see any great urgency. Would you be happy if I put in an expiry time of a 10 days?

Reply:
I will however add a `clear_run_state` function to align with the `entityservice.cache.progress` api.

Reply:
A default of 10 days or more is perfect (configurable even better :) ).
It's mainly to start having best practices when using redis.